specificatie/openapi.yaml

```diff
@@ -648,6 +828,94 @@ components:
               items:
                 $ref: '#/components/schemas/ZoekResultaatHal'
 
+    GeoJsonZoekOpdracht:
+      description: 'Zoekobject om te zoeken op geometrie.'
+      type: object
+      properties:      
+        geometrie:
+          $ref: 'http://schemas.opengis.net/ogcapi/features/part1/1.0/openapi/schemas/polygonGeoJSON.yaml'
+        spatialOperator:
+          type: string
+          description: 'Een uit de set van spatial operators die gedefinieerd zijn in ISO 19125-1'
+          enum:
+#            - bbox
+#            - equals
+#            - disjoint
+            - intersects
+#            - touches
+#            - crosses
+            - within
+            - contains
+#            - overlaps
+#            - beyond
+#            - dwithin
+        tolerantie:
+          type: number
+
+    AdressenZoekObject:
+      type: object
+      description: Een object waarin de benodigde zoek criteria worden opgegeven om naar adressen te zoeken.
+      properties:
+        _geo:
+          $ref: '#/components/schemas/GeoJsonZoekOpdracht'
+        bouwjaar:
+          description: Een aanduiding voor de minimale en/of maximale waarde voor een bouwjaar van een pand op het adres.
+          type: object
+          properties:
```

Comment:
Als een property een object is, definiëren we dit object altijd als component. Anders gaan codegeneratoren hier classes van maken waarbij ze zelf de naam moeten verzinnen. Liever bepalen we zelf hoe componenten/classes heten.

Kan je de min-max component hergebruiken, want zowel bouwjaar als oppervlakte hebben we dit? Het enige verschil zijn de minimum- en maximumwaarden.  Voor oppervlakte zijn er ook minimum- en maximumwaarden (tussen 1 en 999999). Maakt het voor de werking van de API uit als ook voor bouwjaar min: 1 en max: 999999 wordt gebruikt? We hebben denk ik geen gebouwen uit het jaar 0, dus het lijkt me niet erg om niet min: 0 te kunnen opgeven.  Gaat er iets fout als je gebouwen met bouwjaar ≤ 999999 zoekt?

Reply:
OK